analyzers/src/SonarAnalyzer.CSharp/Rules/ForLoopConditionAlwaysFalse.cs

```diff
@@ -120,72 +120,64 @@ private static bool IsLogicalNot(ExpressionSyntax expression, out PrefixUnaryExp
                 && prefixUnaryExpression.OperatorToken.IsKind(SyntaxKind.ExclamationToken);
         }
 
-        /// <summary>
-        /// We try to retrieve the integer value of an expression. If the expression is an integer literal, we return its value, otherwise if
-        /// the expression is an identifier, we attempt to retrieve the integer value the variable was initialized with if it exists.
-        /// </summary>
-        /// <param name="variableNameToIntegerValue">A dictionary mapping variable names to the integer value they were initialized with if it exists</param>
-        /// <param name="expression">The expression for which we want to retrieve the integer value</param>
-        /// <param name="intValue">The output parameter that will hold the integer value if it is found</param>
-        /// <returns>true if an integer value was found for the expression, false otherwise</returns>
-        private static bool GetIntValue(IDictionary<string, int> variableNameToIntegerValue, ExpressionSyntax expression, out int intValue)
+        private static bool DoubleValue(IDictionary<string, double> variableNameToDoubleValue, ExpressionSyntax expression, out double parsedValue)
         {
-            if (ExpressionNumericConverter.TryGetConstantIntValue(expression, out intValue)
+            if (ExpressionNumericConverter.TryGetConstantDoubleValue(expression, out parsedValue)
                 || (expression is SimpleNameSyntax simpleName
-                    && variableNameToIntegerValue.TryGetValue(simpleName.Identifier.ValueText, out intValue)))
+                    && variableNameToDoubleValue.TryGetValue(simpleName.Identifier.ValueText, out parsedValue)))
             {
                 return true;
             }
 
-            intValue = default(int);
+            parsedValue = default;
             return false;
         }
 
         /// <summary>
-        /// Retrieves the mapping of variable names to their integer value from the variable declaration part of a for loop.
+        /// Retrieves the mapping of variable names to their double value from the variable declaration part of a for loop.
         /// This will find the mapping for such cases:
         /// <code>
         /// for (var i = 0;;) {}
         /// </code>
         /// </summary>
-        private static IDictionary<string, int> GetVariableDeclarationMapping(VariableDeclarationSyntax variableDeclarationSyntax)
+        private static IDictionary<string, double> VariableDeclarationMapping(VariableDeclarationSyntax variableDeclarationSyntax)
         {
-            var loopInitializerValues = new Dictionary<string, int>();
+            var loopInitializerValues = new Dictionary<string, double>();
             if (variableDeclarationSyntax != null)
             {
                 foreach (var variableDeclaration in variableDeclarationSyntax.Variables)
                 {
                     if (variableDeclaration.Initializer is EqualsValueClauseSyntax initializer
```

Comment:
We can simplify this to `if (variableDeclaration.Initializer is { } initializer`